test/test_integration_management.cxx

```diff
@@ -884,6 +884,66 @@ TEST_CASE("integration: user management", "[integration]")
         REQUIRE_SUCCESS(resp.ctx.ec);
         REQUIRE(resp.roles.size() > 0);
     }
+
+    SECTION("change user password")
+    {
+        asio::io_context io;
+        auto guard = asio::make_work_guard(io);
+        std::thread io_thread([&io]() { io.run(); });
+        auto user_name = test::utils::uniq_id("newUser");
+
+        // create options
+        auto options_original = couchbase::cluster_options(integration.ctx.username, integration.ctx.password);
+        auto options_outdated = couchbase::cluster_options(user_name, integration.ctx.password);
+        auto options_updated = couchbase::cluster_options(user_name, "newPassword");
+
+        // create new user and upsert
+        couchbase::core::management::rbac::user new_user{ user_name };
+        new_user.display_name = "change_password_user";
+        new_user.password = integration.ctx.password;
+        new_user.roles = {
+            couchbase::core::management::rbac::role{ "admin" },
+        };
+        auto [cluster, ec] = couchbase::cluster::connect(io, integration.ctx.connection_string, options_original).get();
+        auto coreCluster = cluster.core();
+        couchbase::core::operations::management::user_upsert_request upsertReq{};
+        upsertReq.user = new_user;
+        auto upsertResp = test::utils::execute(coreCluster, upsertReq);
+        REQUIRE_SUCCESS(upsertResp.ctx.ec);
+        usleep(100000); // Sleep so server can process
```

Comment:
usleep isn't supported in windows.   Try `std::this_thread::sleep_for(std::chrono::seconds(10)` instead.   It would be nice not to sleep for an arbitrary amount of time, though, if possible.   I wonder if we can ping the server, looking for the user, until they are there?   

That may be worth doing here -- perhaps worth exploring?

Reply:
Thanks David! Added a "wait until user present" to remove the arbitrary wait time.